app/model/RegexRule.scala

```diff
@@ -21,6 +21,7 @@ case class RegexRule(
 
   def toMatch(start: Int, end: Int, block: TextBlock): RuleMatch = {
     val matchedText = block.text.substring(start, end)
+    val replacementText = replacement.map(replacement => regex.replaceAllIn(matchedText, replacement.text))
```

Comment:
This is a bit unweildy.  If I read it correctly, it goes from Option[TS] to Option[String?] only to be turned back into Option[TS] on line 33.

How about a `replaceAllIn` method on the TextSuggestion object which wraps this functionality returning a new TS?

Reply:
Hm, I'd prefer our `TextSuggestion` to avoid having opinions about regular expressions, but you're right, this is a silly dance – we should at least do it in one step, I think. How about –

```scala
val derivedReplacement = replacement.map(replacement =>
      replacement.copy(text = regex.replaceAllIn(matchedText, replacement.text))
    )
```

Reply:
This is definitely better than it was.

If RegexRule objects can have opinions about TextSuggestions, I see no reason the converse shouldn't be true :)